### PR TITLE
Add e2e for remaining quota resources

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Add e2e for quota on replication controller and persistent volume claims to round out missing coverage.

/cc @ncdc

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
I also bumped up default secret count to 10 based on feedback to not hard-code it to 2 since downstream may add more default secrets to a namespace.

File: test/e2e/resource_quota.go [hunk 1]
```diff
@@ -451,6 +522,55 @@ func newTestPodForQuota(name string, requests api.ResourceList, limits api.Resou
...
+				Requests: api.ResourceList{
+					api.ResourceName(api.ResourceStorage): resource.MustParse("1Gi"),
+				},
+			},
+		},
+	}
+}
+
+// newTestReplicationControllerForQuota returns a simple replication controller
+func newTestReplicationControllerForQuota(name string, image string, replicas int) *api.ReplicationController {
```
Comment (pmorie) [inline test/e2e/resource_quota.go]: nit, I am a huge fan of `(name, image string, replicas int)` over `(name string, image string, replicas int)`

**Comment by derekwaynecarr** [comment]
@pmorie nit addressed.

**Comment by derekwaynecarr** [comment]
@pmorie pushed

File: test/e2e/resource_quota.go [hunk 2]
```diff
@@ -451,6 +522,55 @@ func newTestPodForQuota(name string, requests api.ResourceList, limits api.Resou
...
+				Requests: api.ResourceList{
+					api.ResourceName(api.ResourceStorage): resource.MustParse("1Gi"),
+				},
+			},
+		},
+	}
+}
+
+// newTestReplicationControllerForQuota returns a simple replication controller
+func newTestReplicationControllerForQuota(name, image string, replicas int) *api.ReplicationController {
```
Comment (pmorie) [inline test/e2e/resource_quota.go]: sweet relief

**Comment by derekwaynecarr** [comment]
Kicking travis

**Pull request merged**